### PR TITLE
Documentation: PostDiscussionPanel editor component

### PR DESCRIPTION
**Pull request by colorful-tones**
## What? & Why?
Addresses one item in #60358

Adding documentation to existing editor components can help with any of the following:
- encourages knowledge sharing and quicker onboarding for future devs
- supports maintenance and troubleshooting
- mitigates risk

## How?
Add a JSDoc comment to the `PostDiscussionPanel` component and run `npm run docs:build` to populate the `README` with the newly added documents.

### CODE REVIEW

**Comment by Mamaduka** [comment]
I think some of these panel components are becoming absolete with recent refactoring and probably will be deprecated. The question is if we still want to document those.

File: packages/editor/src/components/post-discussion/panel.js [hunk 1]
```diff
@@ -86,6 +86,16 @@ function PostDiscussionToggle( { isOpen, onClick } ) {
 	);
 }
 
+/**
+ * Uses the `useSelect` hook to determine if the discussion panel is enabled.
```
Comment (ntsekouras) [inline packages/editor/src/components/post-discussion/panel.js]: I think we should have a higher level small description of the component and not describe its current implementation, with maybe the exception of mentioning the panel name that can be used to programmatically hide the panel. Something like:

> This component allows to update comment and pingback settings for the current post. Internally there are checks whether the current post has support for the above and if the `discussion-panel` panel is enabled.

**Comment by ntsekouras** [comment]
>I think some of these panel components are becoming absolete with recent refactoring and probably will be deprecated. The question is if we still want to document those.

That's true and if we find components that are only exist for back compat we should deprecate them. Having some high level docs even for them though, until they are removed (if they are removed sometime in the future 😄 ). I think it's fine.

Having said that the specific component in this PR is used in our codebase too.

File: packages/editor/src/components/post-discussion/panel.js (hunk 1 above)
Comment (Mamaduka) [reply packages/editor/src/components/post-discussion/panel.js]: Agreed. The implementation details aren't crucial for similar documents.

**Comment by colorful-tones** [comment]
Thanks for all the guidance and clarification @Mamaduka @ntsekouras 👍 
I streamlined the docs.

**Review by Mamaduka — APPROVED** [review]
Thanks for the follow-up, @colorful-tones!

**Pull request merged**